RecoJets/JetProducers/plugins/CSJetProducer.cc

```diff
@@ -0,0 +1,160 @@
+void CSJetProducer::runAlgorithm( edm::Event & iEvent, edm::EventSetup const& iSetup)
+{
+  // run algorithm
+  if ( !doAreaFastjet_ && !doRhoFastjet_) {
+    fjClusterSeq_ = ClusterSequencePtr( new fastjet::ClusterSequence( fjInputs_, *fjJetDefinition_ ) );
+  } else if (voronoiRfact_ <= 0) {
+    fjClusterSeq_ = ClusterSequencePtr( new fastjet::ClusterSequenceArea( fjInputs_, *fjJetDefinition_ , *fjAreaDefinition_ ) );
+  } else {
+    fjClusterSeq_ = ClusterSequencePtr( new fastjet::ClusterSequenceVoronoiArea( fjInputs_, *fjJetDefinition_ , fastjet::VoronoiAreaSpec(voronoiRfact_) ) );
+  }
+
+  fjJets_.clear();
+  std::vector<fastjet::PseudoJet> tempJets = fastjet::sorted_by_pt(fjClusterSeq_->inclusive_jets(jetPtMin_));
+
+  //Get local rho and rhom map
+  edm::Handle<std::vector<double>> etaRanges;
+  edm::Handle<std::vector<double>> rhoRanges;
+  edm::Handle<std::vector<double>> rhomRanges;
+  
+  iEvent.getByToken(etaToken_, etaRanges);
+  iEvent.getByToken(rhoToken_, rhoRanges);
+  iEvent.getByToken(rhomToken_, rhomRanges);
+
+  //Check if size of eta and background density vectors is the same
+  unsigned int bkgVecSize = etaRanges->size();
+  if(bkgVecSize<1) { throw cms::Exception("WrongBkgInput") << "Producer needs vector with background estimates\n"; }
+  if(bkgVecSize != (rhoRanges->size()+1) || bkgVecSize != (rhomRanges->size()+1) ) {
+    throw cms::Exception("WrongBkgInput") << "Size of etaRanges (" << bkgVecSize << ") and rhoRanges (" << rhoRanges->size() << ") and/or rhomRanges (" << rhomRanges->size() << ") vectors inconsistent\n";
+  }
+
+  
+  //Allow the background densities to change within the jet
+
+  for(fastjet::PseudoJet& ijet : tempJets ) {
+  
+    //----------------------------------------------------------------------
+    // sift ghosts and particles in the input jet
+    std::vector<fastjet::PseudoJet> particles, ghosts;
+    fastjet::SelectorIsPureGhost().sift(ijet.constituents(), ghosts, particles);
+    unsigned long nGhosts=ghosts.size();
+    unsigned long nParticles=particles.size();
+    if(nParticles==0) continue; //don't subtract ghost jets
+    
+    //assign rho and rhom to ghosts according to local eta-dependent map
+    double rho  = 1e-6;
+    double rhom = 1e-6;
+    for (unsigned int j=0;j<nGhosts; j++) {
+      
+      if(ghosts[j].eta()<=etaRanges->at(0) || bkgVecSize==1) {
+        rho = rhoRanges->at(0);
+        rhom = rhomRanges->at(0);
+      } else if(ghosts[j].eta()>=etaRanges->at(bkgVecSize-1)) {
+        rho = rhoRanges->at(bkgVecSize-2);
+        rhom = rhomRanges->at(bkgVecSize-2);
+      } else {
+        for(unsigned int ie = 0; ie<(bkgVecSize-1); ie++) {
+          if(ghosts[j].eta()>=etaRanges->at(ie) && ghosts[j].eta()<etaRanges->at(ie+1)) {
+            rho = rhoRanges->at(ie);
+            rhom = rhomRanges->at(ie);
+            break;
+          }
+        }
+      }
+      double pt = rho*ghosts[j].area();
+      double mass_squared=pow(rhom*ghosts[j].area()+pt,2)-pow(pt,2);
```

Comment:
I didn't actually think about performance... AFAIK the main reason is that `pow` is always double precision ([man pow](https://linux.die.net/man/3/pow)), while `std::pow` will use the precision of its argument, and can be specialised for an integer power ([cppreference.com](http://en.cppreference.com/w/cpp/numeric/math/pow)).
At least until C++11 - looks like C++14 changes it again...